### DISCUSSION TITLE
Running reactomeGSA on multiple resolution cluster separately

### DISCUSSION THREAD

**Discussion by Ankita-1211** [reactome/ReactomeGSA#42] [Q&A] [answered]
Hi 

Thank you for helping me running reactomeGSA for pathway analysis.

I need lil help in understanding this.

The seurat-result.rds file I am using is having multiple resolution cluster information.

my seurat@metadata look like this

<html xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:x="urn:schemas-microsoft-com:office:excel" xmlns="http://www.w3.org/TR/REC-html40">
<head>

<meta name=Generator content="Microsoft Excel">

<style>

</style>
</head>
<body>

  | orig.ident | nCount_RNA | nFeature_RNA | percent.mt | nCount_SCT | nFeature_SCT | SCT_snn_res.0.1 | seurat_clusters | SCT_snn_res.0.2 | SCT_snn_res.0.3 | SCT_snn_res.0.4 | SCT_snn_res.0.5 | SCT_snn_res.0.6 | SCT_snn_res.0.7 | SCT_snn_res.0.8 | SCT_snn_res.0.9 | SCT_snn_res.1.0 | SCT_snn_res.1.1 | SCT_snn_res.1.2 | SCT_snn_res.1.3 | SCT_snn_res.1.4 | SCT_snn_res.1.5
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
AACACACTCGACGAGA | SeuratProject | 4462 | 2071 | 0.054459883 | 6629 | 2093 | 0 | 2 | 1 | 1 | 1 | 1 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 2 | 2
AACAGGGTCAGAATAG | SeuratProject | 19959 | 4112 | 0.091838268 | 8600 | 2649 | 0 | 3 | 1 | 1 | 1 | 1 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 4 | 4

</body>

</html>
 
how can I used this seurat cluster file for running reactomeGSA on each single resolution one by one.

So far when I try to set the indents as Idents(object = data) <- "SCT_snn_res.0.8"

and then running gsva_result <- analyse_sc_clusters(seuset, verbose = TRUE)
it gives me error **Error: Only one identification found: 'integrated_snn_res.0.8'. Please ensure that cell / cluster ids are stored as the primary identification (Ident) of your Seurat object. Clustering has to be performed prior to this pathway analysis.** 

Could you pls guide me in running the reactomeGSA for each resolution cluster separately?

**Comment by jgriss** [comment, marked as answer]
Dear @Ankita-1211 ,

You can find detailed instructions on how to work with `Seurat` objects in the `Seurat` documentation.

Your code sets the identification of all cells to "SCT_snn_res.0.8" (the actual string).

What you rather want is setting it to the clustering results using

```r
Idents(object = data) <- data$`SCT_snn_res.0.8`
```

That way, the *values* of the `SCT_snn_res.0.8` slot are used as the cell identifiers.

Kind regards,
Johannes
